index.js

```diff
@@ -0,0 +1,247 @@
+const express = require("express");
+const xrpl = require("xrpl");
+require("dotenv").config();
+const { Attendify } = require("./attendify");
+const {
+  postToIPFS,
+  ascii_to_hexa,
+  ERR_ATTENDIFY,
+  ERR_IPFS,
+  ERR_NOT_FOUND,
+  ERR_PARAMS,
+  ERR_XRPL,
+} = require("./utils");
+const fs = require("fs");
+
+const app = express();
+let AttendifyLib = new Attendify();
+
+/**
+ * Creates new event,
+ * Uploads its metadata to IPFS
+ * and mints a batch of NFTs for it
+ * @route GET /api/mint
+ * @param {string} walletAddress - The wallet address to mint the NFT tokens to
+ * @param {integer} tokenCount - The number of NFT tokens to mint
+ * @param {string} url - The URL of the event
+ * @param {string} title - The title of the event
+ * @param {string} desc - The description of the event
+ * @param {string} loc - The location of the event
+ * @returns {object} result - An object with the mint result
+ * @throws {Error} If any of the walletAddress, tokenCount, url, title, desc, or loc parameters are missing or have an invalid value
+ */
+app.get("/api/mint", (req, res) => {
+  (async () => {
+    try {
+      const { walletAddress, tokenCount, url, title, desc, loc } =
+        await req.query;
+      if (
+        walletAddress.length == 0 ||
+        tokenCount.length == 0 ||
+        url.length == 0 ||
+        title.length == 0 ||
+        walletAddress.length == 0 ||
+        desc.length == 0 ||
+        loc.length == 0
```

Comment:
When I forgot a paramater or mistyped one, these checks error because the value is actually undefined instead of having a length of 0. (At least on my machine) so they raise the opaque error: “TypeError: Cannot read properties of undefined (reading 'length')” instead of one which tells me which paramaters I messed up.

Reply:
That's a good point, in my case I had to check for empty strings but I'll add check for null values too.

Also regarding comment above: `api/claim` currently returns just status of claim and the list with NFTs for selected event that aren't claimed yet but I suppose you were asking for `minter` parameter that's passed to this endpoint. In current implementation usually it's wallet address for account from `.env` but with modified implementation this could support multiple minter accounts.